tests/react/transition.test.tsx

```diff
@@ -51,12 +52,12 @@ describeWithUseTransition('useTransition', () => {
     )
 
     resolve()
-    await findByText('delayed: 0')
+    expect(await findByText('delayed: 0')).toBeTruthy()
 
-    fireEvent.click(getByText('button'))
-    await waitFor(() => {
+    await userEvent.click(getByText('button'))
+    await waitFor(async () => {
       resolve()
-      getByText('delayed: 1')
+      return expect(await findByText('delayed: 1')).toBeTruthy()
```

Comment:
Are you sure if we need findByText here?

Reply:
Interesting. But you don't need return right?

I'd like to see if this works.
```suggestion
      await waitFor(() => expect(getByText('delayed: 1')).toBeTruthy())
```

Reply:
No luck! 😟 It still fails intermittently. However, wrapping `resolve` within `act` seems to work with this approach ^
<img width="623" alt="image" src="https://user-images.githubusercontent.com/14841132/219228033-2e3223e3-35af-496d-88d3-e5fdb5f998d6.png">

Reply:
Very interesting. We can leave it then. You can remove `return` right?

Reply:
Seems like removing `return` makes it flaky too. I wonder if its because lack of `act` 🤔 

<img width="528" alt="image" src="https://user-images.githubusercontent.com/14841132/219230889-c2b210d7-6cfa-4c69-bcab-5f1d1a9b2527.png">

Reply:
Very very interesting. If `act` helps, it should be better than `return`, I suppose.

Reply:
Oh, I wasn't aware this block is within `waitFor`.